tests/test_torch.py

```diff
@@ -165,26 +175,74 @@ def circuit(reused_param, other_param):
                 zero_state = np.array([1.,0.])
                 final_state = (Rx @ Rz @ Rx @ zero_state)
                 y_true = expZ(final_state)
-                self.assertAlmostEqual(y_eval, y_true, delta=self.tol)
 
-    def test_qnode_array_parameters(self):
-        "Test that QNode can take arrays as input arguments, and that they interact properly with PyTorch."
-        self.logTestName()
+                assert np.allclose(y_eval, y_true, atol=tol, rtol=0)
+
+    def test_qnode_array_parameters_scalar_return(self, qubit_device_1_wire, tol):
+        """Test that QNode can take arrays as input arguments, and that they interact properly with PyTorch."""
 
-        @qml.qnode(self.dev1, interface='torch')
-        def circuit_n1s(dummy1, array, dummy2):
+        @qml.qnode(qubit_device_1_wire, interface='torch')
+        def circuit(dummy1, array, dummy2):
             qml.RY(0.5 * array[0,1], wires=0)
             qml.RY(-0.5 * array[1,1], wires=0)
             return qml.expval(qml.PauliX(0))  # returns a scalar
 
-        @qml.qnode(self.dev1, interface='torch')
-        def circuit_n1v(dummy1, array, dummy2):
+        grad_target = (np.array(1.), np.array([[0.5,  0.43879, 0], [0, -0.43879, 0]]), np.array(-0.4))
+        cost_target = 1.03257
+
+        args = (torch.tensor(0.46), torch.tensor([[2., 3., 0.3], [7., 4., 2.1]]), torch.tensor(-0.13))
+        for i in args:
+            i.requires_grad = True
+
+        def cost(x, array, y):
+            c = torch.as_tensor(circuit(torch.tensor(0.111), array, torch.tensor(4.5)), dtype=torch.float32)
+            if c.shape:
+                c = c[0]  # get a scalar
+            return c +0.5*array[0,0] +x -0.4*y
+
+        cost_res = cost(*args)
+        cost_res.backward()
+
+        assert np.allclose(cost_res.detach().numpy(), cost_target, atol=tol, rtol=0)
+
+        for i in range(3):
+            assert np.allclose(args[i].grad.detach().numpy(), grad_target[i], atol=tol, rtol=0)
+
+    def test_qnode_array_parameters_1_vector_return(self, qubit_device_1_wire, tol):
+        """Test that QNode can take arrays as input arguments, and that they interact properly with PyTorch."""
```

Comment:
docstring is the same as the function above

Reply:
Guys, see my comments in PR #266 related to these three tests

Reply:
Resolved both torch and tfe in the same way